src/firestore/app/appkit.ts

```diff
@@ -236,19 +241,47 @@ export class RoarAppkit {
       const oldVariantId = this.task!.variantId;
       return this.task!.updateTaskParams(newParams)
         .then(() => {
-          return updateDoc(this.user!.userRef, { variants: arrayRemove(oldVariantId) });
+          return updateDoc(this.user!.userRef, {
+            variants: arrayRemove(oldVariantId),
+          });
         })
         .then(() => {
-          return updateDoc(this.user!.userRef, { variants: arrayUnion(this.task!.variantId) });
+          return updateDoc(this.user!.userRef, {
+            variants: arrayUnion(this.task!.variantId),
+          });
         })
         .then(() => {
-          return updateDoc(this.run!.runRef, { variantId: this.task!.variantId });
+          return updateDoc(this.run!.runRef, {
+            variantId: this.task!.variantId,
+          });
         });
     } else {
       throw new Error('This run has not started. Use the startRun method first.');
     }
   }
 
+  /**
+   * Add interaction data for the current trial.
+   *
+   * This method records an interaction event for the current trial and optionally filters
+   * which event types to track. If `requiredEvents` is provided, only events matching those
+   * types will be recorded. Otherwise, all events defined in `trialInteractions` will be tracked.
+   *
+   * The interaction log is stored at the trial level and will be reset after each `writeTrial` call.
+   *
+   * @param {InteractionEvent} interaction - The interaction event to record (e.g., blur, focus).
+   * @param {Array<'blur' | 'focus' | 'fullscreenenter' | 'fullscreenexit'>} [requiredEvents] - Optional list of interaction types to track.
+   * @throws {Error} If the run has not been started.
+   * @method
+   */
+  addInteractions(interaction: InteractionEvent, requiredEvents?: (keyof InteractionSummary<number>)[]) {
```

Comment:
Maybe this should also be called `addInteraction` (singular)

```suggestion
  addInteraction(interaction: InteractionEvent, requiredEvents?: (keyof InteractionSummary<number>)[]) {
```